src/azure-cli/azure/cli/command_modules/vm/_params.py

```diff
@@ -467,19 +467,30 @@ def load_arguments(self, _):
         c.argument('vm_name', arg_type=existing_vm_name, id_part=None)
 
     with self.argument_context('vm image') as c:
-        c.argument('publisher_name', options_list=['--publisher', '-p'])
+        c.argument('publisher_name', options_list=['--publisher', '-p'], help='image publisher')
         c.argument('publisher', options_list=['--publisher', '-p'], help='image publisher')
         c.argument('offer', options_list=['--offer', '-f'], help='image offer')
         c.argument('plan', help='image billing plan')
         c.argument('sku', options_list=['--sku', '-s'], help='image sku')
         c.argument('version', help="image sku's version")
-        c.argument('urn', help="URN, in format of 'publisher:offer:sku:version'. If specified, other argument values can be omitted")
+        c.argument('urn', help="URN, in format of 'publisher:offer:sku:version' or 'publisher:offer:sku:edge_zone:version'. If specified, other argument values can be omitted")
 
     with self.argument_context('vm image list') as c:
         c.argument('image_location', get_location_type(self.cli_ctx))
+        c.argument('edge_zone', options_list=['--edge-zone'], min_api='2020-12-01', help='The name of the edge zone')
+
+    with self.argument_context('vm image list-offers') as c:
+        c.argument('edge_zone', options_list=['--edge-zone'], min_api='2020-12-01', help='The name of the edge zone')
+
+    with self.argument_context('vm image list-skus') as c:
+        c.argument('edge_zone', options_list=['--edge-zone'], min_api='2020-12-01', help='The name of the edge zone')
+
+    with self.argument_context('vm image list-publishers') as c:
+        c.argument('edge_zone', options_list=['--edge-zone'], min_api='2020-12-01', help='The name of the edge zone')
 
     with self.argument_context('vm image show') as c:
         c.argument('skus', options_list=['--sku', '-s'])
+        c.argument('edge_zone', options_list=['--edge-zone'], min_api='2020-12-01', help='The name of the edge zone')
```

Comment:
Could we abstract a reusable `CLIArgumentType` for `--edge-zone` to reduce duplicate attribute definitions?